Simons Recipe Site

simfrisk opened the pull request:
Here is my site:
https://simons-recipe.netlify.app/

I got the main things done but of course I would like to fix it up a bit more but for now I feel happy with the site. 

Things to know:
- The comments that say "#region..." is a way to fold in vscode on my comments to make it more manageable.
- I use a separate css for my color pallet so that is why the names can be like `color: var(--secondary);`  and this will make it easer to add dark mode later och easily change the pallet.
- If the API does not work it will fall back on my "objectRecepie.js" for stand in recipes.

Things I could not get to work
- I could not get the hide corrupted images for the life of me. If you have a solution please share. 
- Certain combinations of filtering can reset the filtering but the main two work together.

Looking forward for your comments!


/Simon

simfrisk committed:
started the site

simfrisk committed:
Getting the css right

simfrisk committed:
css fixes and html

simfrisk committed:
Tried js

simfrisk committed:
margin fix

simfrisk committed:
css fixes for phone

simfrisk committed:
css fixes for phone

simfrisk committed:
Organized and made cleaner code

simfrisk committed:
Small fixes

simfrisk committed:
Small fixes

simfrisk committed:
Small fixes

simfrisk committed:
Got the script workining with the sort

simfrisk committed:
Small fixes

simfrisk committed:
Darkmode

simfrisk committed:
Darkmode

simfrisk committed:
I made a new script to add stuff

simfrisk committed:
fixed reset sort on time

simfrisk committed:
fixed reset sort on time

simfrisk committed:
fixed reset sort on time

simfrisk committed:
fixed the filter on cusine

simfrisk committed:
fixed the filter on time

simfrisk committed:
Got the basics down for the Java Script

simfrisk committed:
new css for buttons to scroll

simfrisk committed:
Made the main buttons and the API to work

simfrisk committed:
Added a ranged slider

simfrisk committed:
Few bugfixes with slider

simfrisk committed:
Few bugfixes with slider

simfrisk committed:
Backup recepies

simfrisk committed:
Backup recepies

simfrisk committed:
Load more recepies function added

simfrisk committed:
bug fixes

simfrisk committed:
Final fixes before the hand in

simfrisk commented:
Yeay!

tildetilde commented:
Code Review: Recipe Library
It’s amazing you’ve built a fully working site with filtering, API fallback, dark mode preparation, and a great visual structure! Thanks for sharing your project and the context behind it — it’s clear you’ve put in solid work. Below is a review based on the current state of your HTML, CSS, and JS setup. I’ve highlighted what’s working well, along with suggestions that could help refine things further (based on your comments).

Strengths 

Clear organization
* Using #region for folding in VSCode is a smart way to manage complexity. It keeps the code more readable, especially in longer CSS files.
* Separating the color palette into its own file is a good move for scalability — it’ll make switching themes (like adding dark mode) much easier.
Solid layout and responsiveness
* You’ve set up a responsive grid for the recipe cards using grid-template-columns and clamp(), which adapts well across screen sizes.
* The media queries are well-targeted and reduce visual clutter on smaller devices.
Fallback handling
* Including objectRecepies.js as a fallback if the API fails is thoughtful and improves 
Infiinite scroll
* It’s cool that you made it work!
Price slider
* It’s a really nice touch!

Suggestions & Fixes
Handling broken images
One way to hide corrupted images is to add this in your JavaScript where you create or load image elements (innerHTML): <img src="..." onerror="this.style.display='none’;”>. If you'd rather show a fallback image, you could replace the src instead.

Filter logic
I’m not sure why this is happening, but it could be that each filter (cuisine, time, price, search) seems to independently call renderRecipes() with the full array (allRecipes), rather than applying filters cumulatively. Maybe applying filters cumulative would help?

Reset
I think it would be nice to add an option to reset filters, in order to come back to all recipes again. And I did not really get the Empthy sorting option, it seems to render new recipes – but it is under sorting. Maybe that should be called ‘Get new recipes’? 

Lastly, congratulations on building a very cool site! 🤗

HIPPIEKICK reviewed (changes requested):
Nice solution for the buttons, makes it less cluttered 👍 Would be nice if you had a button with All kitchens or a button to clear the filters.

Also, I found a bug. When choosing Italy, it shows me “There are no recipes…” but it’s also showing some other recipes. I think it would be better to clear the DOM if no recipes can be displayed for a specific filter.

And another bug: Your infinite scrolling doesn’t work together with your filters (after picking a filter and scrolling down, it shows me all kinds of cuisines even though I filtered on e.g. American).

Loving the range slider as well, great idea! However, I found a little CSS bug: You should put a fixed width on the descriptive text next to the range slider, otherwise the search bar jumps when the text element becomes wider.

Your functions and variables have descriptive names, keep this up!

Some things to fix in order to be approved but please be proud of all you’ve accomplished ⭐

PS. Small thing, but double check your spelling (backupRecepies()). Also, what does this do? 🤔 👇
`recipesData = data.recipes || backupRecepies().recipes`

I noticed that you've done most of the stretch goals but just want to clarify that for VG you need to implement localStorage.

simfrisk committed:
fixed all btn

simfrisk committed:
italy fix

simfrisk committed:
spelling

simfrisk committed:
fixed all except scroll

simfrisk commented:
@HIPPIEKICK 

Hi Matilda! Thank you for all your great feedback. Really happy to see that you don’t just jump through the reviews but actually take the time to look through them!

1. Italy button problem:
I think I fixed the Italy problem! Don’t know what I did, but it seemed to work.

2. The infinite scroll and filtering bug:
I tried to fix this, but it drove me crazy, so eventually, I gave up. Since I still think I covered enough criteria for the VG grade, I’d like to focus on the current project. Might get back to it later. Anyway, I commented it out for now.

3. The slider label text alignment:
I want it like this on purpose since I’d rather make the text jump than have the next input jump down.

4. backupRecepies() spelling mistake:
Fixed that.

5. recipesData = data.recipes || backupRecepies().recipes explanation:
Well, I thought this was how local storage was supposed to be done, but after rereading the instructions, I realized that local storage is supposed to be fetched and cached, not hardcoded into the JavaScript. So, this is a backup of a bunch of recipes in case the API is down or the max fetch request limit is reached.
recipesData = data.recipes || backupRecepies() means: if data.recipes exists, fetch from the API; if not, run the backup script, which is in its own JS file.

6. Correct LocalStorage implementation:
Now it should be on par with the instructions. To be honest, I got a bit more AI help on this than usual, but I think I got the gist of it.

HIPPIEKICK reviewed (approved):
Really good job with these improvements, and thanks for taking the time to document your new changes as well.

PS. I meant something like this for the slider 👇 Take it or leave it 😉 

https://github.com/user-attachments/assets/09275272-0775-4d13-8f1b-fc6f7c6eb7f9
